src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/CcdAuthService.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.config.JurisdictionToUserMapping;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.Credential;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+@Service
+class CcdAuthService {
+    private final AuthTokenGenerator s2sTokenGenerator;
+    private final IdamClient idamClient;
+    private final JurisdictionToUserMapping users;
+
+    public CcdAuthService(AuthTokenGenerator s2sTokenGenerator,
+                          IdamClient idamClient,
+                          JurisdictionToUserMapping users) {
+        this.s2sTokenGenerator = s2sTokenGenerator;
+        this.idamClient = idamClient;
+        this.users = users;
+    }
+
+    CcdAuthInfo authenticateForJurisdiction(String jurisdiction) {
+        String sscsToken = s2sTokenGenerator.generate();
+        Credential user = users.getUser(jurisdiction);
+        String userToken = idamClient.authenticateUser(user.getUsername(), user.getPassword());
+        UserDetails userDetails = idamClient.getUserDetails(userToken);
```

Comment:
Do we get a null token if we are not able to authenticate ?

Reply:
yup, it will through a null pointer and be caught later on.
I have not covered the exception cases yet due to the hurry to unblock this card.
As mentioned Ive got the other card to complete this.

Also the error handling has to be thought out correctly so we ack messages that will always be errors since its pointless letting it appear in the queue again.

Reply:
err, no I would expect a FeignException to be thrown
which will have a status code on there,
just needs to be logged when that's caught eventually